rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java

```diff
@@ -216,10 +221,7 @@ private void emitEvents(List<Transaction> addedPendingTransactions) {
             return Optional.empty();
         }
 
-        return txsaccount
-                .stream()
-                .filter(t -> t.getNonceAsInteger().equals(next))
-                .findFirst();
+        return txsaccount.stream().filter(t -> t.getNonceAsInteger().equals(next)).findFirst();
     }
 
     private TransactionPoolAddResult internalAddTransaction(final Transaction tx) {
```

Comment:
It would make sense for transactions that fail to pass even the most basic checks (ie: `!transactionIsValid`) to subtract from the virtual gas available too, right?

Reply:
I'm not sure about this.

IMO if a transaction is not valid it should not be reducing the virtualGas of the account, as it is intended to measure / protect the "execution cost" and this transaction is not going to be executed at all.

Let's wait for @SergioDemianLerner opinion, though.

Reply:
Mhm. OK. So only when your transaction is to be executed you do reduce the virtual gas... that's better for the gas analogy I think. Let's go with that, yeah.